sklearn/ensemble/tests/test_weight_boosting.py

```diff
@@ -328,80 +347,92 @@ def fit(self, X, y, sample_weight=None):
-        dense_results = dense_classifier.staged_predict(X_test)
-        for sprase_res, dense_res in zip(sparse_results, dense_results):
-            assert_array_equal(sprase_res, dense_res)
-
-        # staged_predict_proba
-        sparse_results = sparse_classifier.staged_predict_proba(X_test_sparse)
-        dense_results = dense_classifier.staged_predict_proba(X_test)
-        for sprase_res, dense_res in zip(sparse_results, dense_results):
-            assert_array_almost_equal(sprase_res, dense_res)
-
-        # staged_score
-        sparse_results = sparse_classifier.staged_score(X_test_sparse, y_test)
-        dense_results = dense_classifier.staged_score(X_test, y_test)
-        for sprase_res, dense_res in zip(sparse_results, dense_results):
-            assert_array_equal(sprase_res, dense_res)
-
-        # Verify sparsity of data is maintained during training
-        types = [i.data_type_ for i in sparse_classifier.estimators_]
-
-        assert all([(t == csc_matrix or t == csr_matrix) for t in types])
-
-
-def test_sparse_regression():
+    X_train_sparse = sparse_container(X_train)
+    X_test_sparse = sparse_container(X_test)
+
+    # Trained on sparse format
+    sparse_classifier = AdaBoostClassifier(
+        estimator=CustomSVC(probability=True),
+        random_state=1,
+        algorithm="SAMME",
+    ).fit(X_train_sparse, y_train)
+
+    # Trained on dense format
+    dense_classifier = AdaBoostClassifier(
+        estimator=CustomSVC(probability=True),
+        random_state=1,
+        algorithm="SAMME",
+    ).fit(X_train, y_train)
+
+    # predict
+    sparse_results = sparse_classifier.predict(X_test_sparse)
+    dense_results = dense_classifier.predict(X_test)
+    assert_array_equal(sparse_results, dense_results)
+
+    # decision_function
+    sparse_results = sparse_classifier.decision_function(X_test_sparse)
+    dense_results = dense_classifier.decision_function(X_test)
+    assert_array_almost_equal(sparse_results, dense_results)
+
+    # predict_log_proba
+    sparse_results = sparse_classifier.predict_log_proba(X_test_sparse)
+    dense_results = dense_classifier.predict_log_proba(X_test)
+    assert_array_almost_equal(sparse_results, dense_results)
+
+    # predict_proba
+    sparse_results = sparse_classifier.predict_proba(X_test_sparse)
+    dense_results = dense_classifier.predict_proba(X_test)
+    assert_array_almost_equal(sparse_results, dense_results)
+
+    # score
+    sparse_results = sparse_classifier.score(X_test_sparse, y_test)
+    dense_results = dense_classifier.score(X_test, y_test)
+    assert_array_almost_equal(sparse_results, dense_results)
+
+    # staged_decision_function
+    sparse_results = sparse_classifier.staged_decision_function(X_test_sparse)
+    dense_results = dense_classifier.staged_decision_function(X_test)
+    for sprase_res, dense_res in zip(sparse_results, dense_results):
+        assert_array_almost_equal(sprase_res, dense_res)
```

Comment:
While we are at it, let's fix the typo: `sprase` => `sparse`.

Reply:
Furthermore, the names "sparse_results" and "sparse_res" are confusing. Those are not sparse out datastructures but results of a classifier that fits and predicts on sparse inputs datastructures.

I think we should rename those to `dense_clf_results` / `sparse_clf_results` instead (and similarly for the "_res" variables).